core/services/workflows/v2/engine.go

```diff
@@ -494,10 +504,20 @@ func (e *Engine) startExecution(ctx context.Context, wrappedTriggerEvent enqueue
 		e.lggr.Debugw("User workflow execution result", "result", result.GetValue(), "err", result.GetError())
 	}
 
+	if len(result.GetError()) > 0 {
+		executionStatus = store.StatusErrored
+		e.metrics.UpdateWorkflowErrorDurationHistogram(ctx, int64(executionDuration.Seconds()))
+		executionLogger.Errorw("Workflow execution failed", "status", executionStatus, "durationMs", executionDuration.Milliseconds())
```

Comment:
Optionally we could emit the error as a user log (i.e. push to userLogChan?). We can do it separately.